### ISSUE TITLE
feat(ci): integrate Windows

### ISSUE THREAD

**Issue by jef** [beetbox/beets#3613]
### Use case

From https://github.com/beetbox/beets/pull/3602#issuecomment-639520863

We'd like to get rid of AppVeyor and in favor of building Windows within GitHub Actions.

### Solution

Get `windows-latest` part of the matrix that GitHub Actions runs. Currently seeing encoding errors from `tox` when running.

**Comment by jtpavlock** [comment]
So to further @sampsyo's suspicion that the error deals with tox, I bypassed tox in favor of just calling pytest directly in the workflow and the error disappeared. However, for some reason, there are a couple of tests that fail in the github action that don't fail in appveyor.

**Comment by sampsyo** [comment]
Mysterious on both counts. If the problem appears to be encoding nonsense, I can see two paths forward: actually fixing issues related to insufficiently flexible encoding tolerance, or instructing Tox to set locale variables that force a UTF-8 encoding.

**Comment by jtpavlock** [comment]
So, I believe this is blocked until we drop python 2.7 support.

Let me explain,

There was a previous tox error for python2.7 that was reported (and commented on by our very own @sampsyo) https://github.com/tox-dev/tox/issues/1234. This fix was merged with tox v3.8.4
However, I did some testing, and it seems tox fails on Windows for all versions >=3.8.4 for python 3.8 (and possible other versions of python3). I reported this https://github.com/tox-dev/tox/issues/1550#issuecomment-661381136 
I'm guessing this isn't just a coincidence.

What this means, is that we can't fix this as 3.8.3- fails on python2.7, but 3.8.4+ fails on python3

I'm not sure what appveyor does to prevent this error, as I'm seeing this error on my native Windows machine. 

Once we drop support for python2.7, we can just tell the github action to target 3.8.3 and call it a day.
